lib/streamlit/runtime/secrets.py

```diff
@@ -275,13 +254,12 @@ def _parse_toml_file(self, path: str) -> tuple[Mapping[str, Any], bool]:
 
             secrets.update(toml.loads(secrets_file_str))
         except (TypeError, toml.TomlDecodeError) as ex:
-            error_msg = (
+            msg = (
                 secret_error_messages_singleton.get_error_parsing_file_at_path_message(
                     path, ex
                 )
             )
-            self._print_exception_if_not_suppressed(error_msg)
-            raise
+            raise StreamlitSecretNotFoundError(msg)
```

Comment:
nit: You can do `raise StreamlitSecretNotFoundError(msg) from ex` here to provide even more info on what went wrong.